### PR TITLE
Conditionally render Auth with magic link by accepting a prop from th…

### PR DESCRIPTION
**Pull request by Nazeeh21**
## What kind of change does this PR introduce?
**Signing with magic link** will render conditionally by accepting props from the user

Bug fix, feature, docs update, ...
*Feature*

## What is the current behavior?
**Sign in with magic link** renders every time and unable to hide it in case user don't want this behavior

Please link any relevant issues here.
#246

## What is the new behavior?
Hide magic link conditionally.

### CODE REVIEW

**Comment by Nazeeh21** [comment]
@MildTomato, is there any changes I need to do in this PR?

File: src/components/Auth/Auth.tsx [hunk 1]
```diff
@@ -52,6 +52,7 @@ export interface Props {
   view?: ViewType
   redirectTo?: RedirectTo
   onlyThirdPartyProviders?: boolean
+  withMagicLink?: boolean
```
Comment (MildTomato) [inline src/components/Auth/Auth.tsx:55]: shall we just call it `magicLink` ?

Comment (Nazeeh21) [reply src/components/Auth/Auth.tsx]: Sure! I have changed it. Please do let me know of any other changes 😃 .

**Comment by MildTomato** [comment]
looks great @Nazeeh21 🔥

**Pull request merged**